smoke_tests/sns_create_topic_test.go

```diff
@@ -0,0 +1,256 @@
+	sdkConfig, _ := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+	sdkResponse, err := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+
+	// Should success
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Get created topic
+	listTopicsXML := struct {
+		Action  string `xml:"Action"`
+		Version string `xml:"Version"`
+	}{
+		Action:  "ListTopics",
+		Version: "2012-11-05",
+	}
+	e := httpexpect.Default(t, server.URL)
+	r := e.POST("/").
+		WithForm(listTopicsXML).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+	r2 := app.ListTopicsResponse{}
+	xml.Unmarshal([]byte(r), &r2)
+	assert.Equal(t, 1, len(r2.Result.Topics.Member))
+	assert.Contains(t, r2.Result.Topics.Member[0].TopicArn, topicName)
+}
+
+func Test_CreateTopicV1_json_existant_topic(t *testing.T) {
+	server := generateServer()
+	defer func() {
+		server.Close()
+		utils.ResetResources()
+	}()
+
+	// Prepare existant topic
+	topicName := "new-topic-1"
+	sdkConfig, err := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+	sdkResponse, _ := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Target test: create topic with same name
+	sdkResponse, err = snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+
+	// Should success
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Topic should not be duplicated
+	listTopicsXML := struct {
+		Action  string `xml:"Action"`
+		Version string `xml:"Version"`
+	}{
+		Action:  "ListTopics",
+		Version: "2012-11-05",
+	}
+	e := httpexpect.Default(t, server.URL)
+	r := e.POST("/").
+		WithForm(listTopicsXML).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+	r2 := app.ListTopicsResponse{}
+	xml.Unmarshal([]byte(r), &r2)
+	assert.Equal(t, 1, len(r2.Result.Topics.Member))
+	assert.Contains(t, r2.Result.Topics.Member[0].TopicArn, topicName)
+}
+
+func Test_CreateTopicV1_json_add_new_topic(t *testing.T) {
+	server := generateServer()
+	defer func() {
+		server.Close()
+		utils.ResetResources()
+	}()
+
+	// Prepare existant topic
+	topicName := "new-topic-1"
+	sdkConfig, err := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+	sdkResponse, _ := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Target test: create topic with same name
+	topicName2 := "new-topic-2"
+	sdkResponse, err = snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName2,
+	})
+
+	// Should success
```

Comment:
Is this comment still relevant?

Reply:
Yes, creating a new topic with another name should be successful.

Because CreateTopic returns success even for topics with duplicate names, I wanted to make sure that topics with different names were being added just to be sure.
Do you think this is an excessive testing pattern?